go/adbc/driver/flightsql/flightsql_statement.go

```diff
@@ -279,7 +279,9 @@ func (s *statement) ExecutePartitions(ctx context.Context) (*arrow.Schema, adbc.
 	out.NumPartitions = uint64(len(info.Endpoint))
 	out.PartitionIDs = make([][]byte, out.NumPartitions)
 	for i, e := range info.Endpoint {
-		data, err := proto.Marshal(e)
+		partition := proto.Clone(info).(*flight.FlightInfo)
+		partition.Endpoint = []*flight.FlightEndpoint{e}
+		data, err := proto.Marshal(partition)
```

Comment:
why do this? What if a server provides multiple optional endpoints for a given partition so a consumer can use a geographically closer one? shouldn't we let that be propagated?

Reply:
each endpoint gets serialized into a separate partition

Reply:
gotcha, i was flipping their meaning